gstd/gstd_http.c

```diff
@@ -311,16 +334,36 @@ server_callback (SoupServer * server, SoupMessage * msg,
   g_return_if_fail (msg);
   g_return_if_fail (data);
 
-  session = GSTD_SESSION (data);
-  g_return_if_fail (session);
+  GstdHttp *self = GSTD_HTTP (data);
+  session = self->session;
+
+  struct gstd_http_request *data_request;
+  data_request =
+      (struct gstd_http_request *) malloc (sizeof (struct gstd_http_request));
+
+  data_request->msg = msg;
+  data_request->server = server;
+  data_request->session = session;
+  data_request->path = path;
+  if (query) {
+    data_request->query = g_hash_table_ref (query);
+  } else {
+    data_request->query = query;
+  }
 
   soup_message_headers_append (msg->response_headers,
       "Access-Control-Allow-Origin", "*");
   soup_message_headers_append (msg->response_headers,
       "Access-Control-Allow-Headers", "origin,range,content-type");
   soup_message_headers_append (msg->response_headers,
       "Access-Control-Allow-Methods", "PUT, GET, POST, DELETE");
-  do_request (server, msg, query, session, path);
+
+  soup_server_pause_message (server, msg);
+  /*catch error */
+  if (!g_thread_pool_push (self->pool, (gpointer) data_request, NULL)) {
```

Comment:
Request in the configure.ac file to use **libsoup-2.4 >= 2.32**. Prior to this version, this didn't return success status.
https://developer.gnome.org/glib/stable/glib-Thread-Pools.html#g-thread-pool-push